File: app/src/main/java/com/nilhcem/mobilization/data/app/model/Room.java
package com.nilhcem.mobilization.data.app.model;

import android.support.annotation.NonNull;

public enum Room {

    NONE(0, ""),
    ROOM_1(1, "Rndity"),
    ROOM_2(2, "Ericpol"),
    ROOM_3(3, "Mobica"),
    ROOM_4(4, "SEQR"),
    ROOM_5(5, "TomTom");

    public final int id;
    public final String label;

    Room(int id, String label) {
        this.id = id;
        this.label = label;
    }

    public static Room getFromId(int id) {
        for (Room room : Room.values()) {
            if (room.id == id) {
                return room;
            }
        }
        return NONE;
    }

    public static Room getFromLabel(@NonNull String label) {
        for (Room room : Room.values()) {
            if (label.equals(room.label)) {
                return room;
            }
        }
        return NONE;
    }
}

File: scraper/src/main/kotlin/com/nilhcem/mobilization/scraper/model/output/Room.kt
package com.nilhcem.mobilization.scraper.model.output

// http://2016.mobilization.pl/api/venues.json
enum class Room(val id: Int, val apiId: String, val roomName: String) {

    NONE(0, "", ""),
    ROOM_1(1, "venue-rndity", "Rndity"),
    ROOM_2(2, "venue-ericpol", "Ericpol"),
    ROOM_3(3, "venue-mobica", "Mobica"),
    ROOM_4(4, "venue-seqr", "SEQR"),
    ROOM_5(5, "venue-tomtom", "TomTom");

    companion object {
        fun getByApiId(apiId: String) = Room.values().filter { apiId == it.apiId }.getOrElse(0, { NONE })
    }
}
